optuna/study/_optimize.py

```diff
@@ -210,7 +208,10 @@ def _run_trial(
         thread.start()
 
     try:
-        value_or_values = func(trial)
+        value_or_values = cast(
+            Optional[Union[float, List[float]]],
+            func(trial),
+        )
```

Comment:
Could you add a comment why the `cast` is needed?

Reply:
I tried to explain in 5c57e2b. Please give me a feedback. :pray:

Reply:
Thanks, are we aligned that the error, which is raised by mypy is as follows, unless casted?

```
optuna/study/_optimize.py:216: error: Incompatible types in assignment (expression has type "Union[float, Sequence[float]]", variable has type "Union[float, List[float], None]")
```

It seems like you can change the type hint of `value_or_values` at its declaration (at line 195) from `List`, to `Sequence` as an alternative to the `cast`. Please correct me if I'm wrong.

Reply:
So sorry, you're totally  right... I changed the type of `value_or_values` using `Sequence`.